### ISSUE TITLE
valid-jsdoc tests failing with doctrine@<0.7.1

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#4545]
ESLint at master (v1.10.1)
Doctrine 0.7.0 (`^0.7.0` is specified in ESLint's package.json)

I'm assuming this is more likely a Doctrine problem, but I wanted to create an issue here in case anyone else had the same problem.

Three `valid-jsdoc` unit tests are failing on my machine (and not on Travis) with a reported ESLint error of "JSDoc syntax error":

```js
// valid:
        {
            code: "/** Foo \n@return Foo\n */\nfunction foo(){}",
            options: [{ requireReturnType: false }]
        },

// invalid:
        {
            code: "/** Foo \n@return Foo\n */\nfunction foo(){}",
            options: [{ prefer: { "return": "return" }}],
            errors: [{
                message: "Missing JSDoc return type.",
                type: "Block"
            }]
        },
        {
            code: "/** Foo \n@return sdf\n */\nfunction foo(){}",
            options: [{
                prefer: { "return": "return" },
                requireReturn: false
            }],
            errors: [{
                message: "Unexpected @return tag; function has no return statement.",
                type: "Block"
            }]
        },
```

**Comment by eslintbot** [comment]
@platinumazure Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
Please also share the doctrine version you have installed.
FYI, some test would fail if you dont have latest `doctrine`.

**Comment by platinumazure** [comment]
@gyandeeps 0.7.0, also edited into original post. I see there's a 0.7.1 now, do I need to get that in? I did run `npm prune && npm install` a little bit ago so I'm surprised that didn't put 0.7.1 on my machine...

**Comment by gyandeeps** [comment]
Yes, you need `0.7.1`.
Run `npm update`.

**Comment by platinumazure** [comment]
Understood. Thought of `npm update` only after my last post.

If we really rely on 0.7.1, we should note that in our package.json. Let me know if I should write a PR for that.

**Comment by gyandeeps** [comment]
Sure, You can send one by referencing this issue.
Start your commit message with "Upgrade:".

**Comment by platinumazure** [comment]
@gyandeeps Written as #4546. Thanks!
